src/app/components/AutoComplete.svelte

```diff
@@ -0,0 +1,188 @@
+      key !== keyboardKey.Up &&
+      key !== keyboardKey.Enter
+    )
+      return;
+    if (key == keyboardKey.Down) {
+      prevIndex = index;
+      index += 1;
+    } else if (key == keyboardKey.Up) {
+      if (index == -1) {
+        index = -1;
+      } else {
+        prevIndex = index;
+        index -= 1;
+      }
+    } else if (key == keyboardKey.Enter) {
+      selected = filetred[index].language;
+      subtext = filetred[index].bcp47Tag;
+      show = false;
+    }
+    const allItems = [
+      ...document.getElementsByClassName("autocomplete__suggestion-item"),
+    ];
+
+    index = mod(index, allItems.length);
+    document
+      .getElementsByClassName("autocomplete__suggestion-item")
+      [index].classList.add("autocomplete-active");
+    if (prevIndex !== -1) {
+      document
+        .getElementsByClassName("autocomplete__suggestion-item")
+        [prevIndex].classList.remove("autocomplete-active");
+    }
+  }
+</script>
+
+<style>
+  .autocomplete {
+    position: relative;
+    width: fit-content;
+  }
+  .autocomplete__label {
+    font-family: var(--main-font), sans-serif;
+    font-weight: bold;
+    font-size: var(--xs);
+  }
+  .autocomplete__input {
+    font-family: var(--secondary-font), sans-serif;
+    font-size: var(--m);
+    padding: var(--sb-xs);
+    border-radius: var(--sb-xs);
+    border-width: 1px;
+    border-color: rgba(0, 0, 0, 0.2);
+  }
+  .autocomplete__suggestion-list {
+    position: absolute;
+    padding: 0 0 var(--sb-xs) 0;
+    margin: 0;
+    width: 100%;
+    background: white;
+    box-shadow: 0px var(--sb-xs) var(--sb-m) rgba(0, 0, 0, 0.2);
+    border-radius: 0 0 var(--sb-xs) var(--sb-xs);
+    list-style-type: none;
+    text-decoration: none;
+    z-index: 5;
+  }
+  .autocomplete__suggestion-item {
+    font-size: var(--m);
+    font-style: bold;
+    padding: var(--sb-xs) var(--sb-s);
+    overflow: hidden;
+    border-right: solid var(--sb-s) rgba(0, 0, 0, 0);
+  }
+  .autocomplete__suggestion-item:hover {
+    cursor: pointer;
+    background-color: var(--gray-highlight);
+  }
+  .autocomplete__subtext {
+    font-family: var(--mono-font), monospace;
+    color: var(--gray);
+    font-size: var(--xxs);
+  }
+  /*In Sevelte, dynamic class will only work if it's a global style*/
+  :global(.autocomplete-active) {
+    margin: 0;
+    background-color: var(--gray-highlight);
+  }
+</style>
+
+<svelte:window on:keydown={handleKeydown} />
+<div class="autocomplete">
+  {#if label !== ''}
+    <p class="autocomplete__label">{label}</p>
+  {/if}
+  <input
+    class="autocomplete__input"
+    value={selected}
+    type="text"
+    on:input={(event) => onChange(event)}
+    data-cy="autocomplete-input" />
+  {#if show}
+    <ul
+      class="autocomplete__suggestion-list"
+      use:clickOutside={closeSuggestion}>
+      {#each filetred as result}
+        <li
+          class="autocomplete__suggestion-item"
+          data-cy="autocomplete-suggestions">
```

Comment:
Use the index to match with the currently selected class (see: https://svelte.dev/tutorial/each-blocks)

```suggestion
      {#each filetred as result, i}
        <li
          class="autocomplete__suggestion-item"
          class:autocomplete--active={index == i}
          data-cy="autocomplete-suggestions"
```